games-strategy/globulation2/patches/globulation2-0.9.4.4.patchset

```diff
@@ -514,5 +514,52 @@ index d0ca0ef..dd6e33b 100644
          env.Clone()
      except AttributeError:
 -- 
-2.30.2
+2.36.1
+
+
+From dfe7b13c5b01f869f12880c7f9391fab19b1bb43 Mon Sep 17 00:00:00 2001
+From: Victor Gamper <victor@wenzeslaus.de>
+Date: Sat, 27 Aug 2022 15:19:20 +0200
+Subject: globulation2: moved config folder to B_USER_SETTINGS_DIRECTORY
+
+
+diff --git a/libgag/src/FileManager.cpp b/libgag/src/FileManager.cpp
+index 760a04e..13bdadb 100644
+--- a/libgag/src/FileManager.cpp
++++ b/libgag/src/FileManager.cpp
+@@ -64,16 +64,32 @@
+ #endif
+ #endif
+ 
++#ifdef __HAIKU__
++#include <FindDirectory.h>
++#include <fs_info.h>
++#endif
++
+ namespace GAGCore
+ {
+ 	FileManager::FileManager(const char *gameName)
+ 	{
+ 		#ifndef WIN32
++		#ifndef __HAIKU__
+ 		const char *homeDir = getenv("HOME");
++		#else
```

Comment:
Won't this affect the windows build? Maby `#elif defined(__HAIKU__)` ?

Reply:
it shouldn't, the `#ifndef WIN32` part is in the original aswell, but I had some problems with the newlines created being added into the source file, so this line is in there aswell.
I could try and set up a windows build environment to check that though

Reply:
I redid the patch and replaced `#ifndef WIN32` with `#if !(defined(WIN32) || defined(__HAIKU__))`  . The if clause contains the code that adds the unix folders to the path, I added a Haiku part above it that adds the config directory to the path instead

Reply:
No time to check this, but it still looks like a bit overkill? Maybe someone from the haikuports team can chime in?

Reply:
It doesn't look right to me. I don't think you need to replace the `#ifndef WIN32`.  Adding the `|| defined(__HAIKU__)` part will exclude this from Linux and other operating systems.

Although it's not required, I probably would've used the C++ version of [`find_directory()`](https://www.haiku-os.org/legacy-docs/bebook/BDirectory.html#find_directory).  This simplifies things because you can pass `NULL` to get the boot volume instead of looking it up.